### PR TITLE
Premium Blocks: Handle core/cover block

### PR DESCRIPTION
**Pull request by retrofox**
This PR implements the upgrade banner for `core/cover` blocks. For this purpose, it adds a context that allows to pass data through the components. 

Fixes #

#### Changes proposed in this Pull Request:

* Add Upgrade feature to core/cover block

#### Does this pull request change what data or activity we track or use?

no

#### Testing instructions:

* Apply both patches in your sandbox: D47508-code and D46955-code
* Compile these files and update your sandbox with these changes
* Test in a Simple site with a Free Plan
* Sandbox the testing site.
* Add a `core/cover` block to the editor
* Confirm that the block doesn't show the banner in the editor canvas

* Try to upload a video file either dropping on the media placeholder area or uploading by clicking on the `Upload` button. Confirm it shows the upgrade banner.

* Try to set a background. It should work as expected.

* Try to replace the background using the `Replace` control button from the block toolbar. It should act in the similar way that before.

### CODE REVIEW

File: extensions/shared/premium-blocks/with-upgrade-banner.jsx [hunk 1]
```diff
@@ -1,40 +1,61 @@
...
 		}
 
+		const isDualMode = isStillUsableWithFreePlan( props?.name );
+
+		const [ isVisible, setIsVisible ] = useState( ! isDualMode );
+
+		// Hide Banner when block changes its attributes (dual Mode).
+		useEffect( () =>
+			setIsVisible( ! isDualMode )
+			, [ props.attributes, setIsVisible, isDualMode ]
```
Comment (obenland) [inline extensions/shared/premium-blocks/with-upgrade-banner.jsx:33]: This line formatting is a bit confusing, could that be changed to

```suggestion
		useEffect(
			() => setIsVisible( ! isDualMode ),
			[ props.attributes, setIsVisible, isDualMode ]
```

**Comment by cpapazoglou** [comment]
I followed the **Testing instructions** and I applied the D46955 patch. Unfortunately, I can't get it working. I am getting the following when uploading an `mp4` or `mov`



Is there something I am missing in my build / environment maybe?

Also, it seems that the other premium blocks, now don't have the button

File: extensions/shared/premium-blocks/with-upgrade-banner.jsx (hunk 1 above)
Comment (retrofox) [reply extensions/shared/premium-blocks/with-upgrade-banner.jsx]: just fyi, we are going to make a whole tidying code once this PR gets merged with the feature branch.

Comment (retrofox) [reply extensions/shared/premium-blocks/with-upgrade-banner.jsx]: sorry I didn't realize it was a commit suggestion.

**Comment by retrofox** [comment]
> I followed the **Testing instructions** and I applied the D46955 patch. Unfortunately, I can't get it working. I am getting the following when uploading an `mp4` or `mov`

Sorry, yes. Testing instructions are not totally clear and tricky. Let me try to change that.

> 
> 

Addressed here https://github.com/Automattic/jetpack/pull/16688/commits/80c0a6a1c84d9ea48d57de4b26658fea4d846549. Forgot to include the plan store.

**Comment by retrofox** [comment]
> I followed the **Testing instructions** and I applied the D46955 patch. Unfortunately, I can't get it working. I am getting the following when uploading an `mp4` or `mov`

Thanks @cpapazoglou for your help here.
Basically, while we develop the backend side, you can test this approach applying D47508-code and D46955-code to your sandbox, and also compile and apply these changes there as well.

**Review by cpapazoglou — APPROVED** [review]
Thanks @retrofox for updating the testing instructions! It is definitely working as expected now.

1) Uploading a video or drag n drop results to
image

2) Clicking the `Upgrade` button redirects me to checkout. Checking out redirects to the editor and I can upload a video


LGTM!

**Pull request merged**